### PR TITLE
Implement warmup with mod (DT)

### PR DESCRIPTION
**Pull request by vunyunt**

### CODE REVIEW

File: server/api.ts [hunk 1]
```diff
@@ -772,10 +775,7 @@ router.postAsync("/streamer", ensure.isStreamer, async (req, res) => {
 });
 
 /**
- * DELETE /api/streamer
- * Removes the current streamer
- * Params:
- *  - match: the _id of the match
+ * DELETE /api/streamer| "HT"
```
Comment (cory2067) [inline server/api.ts:778]: ?

**Review by cory2067 — COMMENTED** [review]
also should update the comment describing the expected args for POST /warmup, or even better, make the request body typed with typescript (see api/maps.ts for examples)

and should fix the broken test

File: server/api.ts (hunk 1 above)
Comment (vunyunt) [reply server/api.ts]: Oops probably a mis-paste, restored in 69f43497be6be94b4bcceb648c6491780279a936

**Comment by vunyunt** [comment]
TS-typed body sounds good, I'll try that

File: server/api.ts [hunk 2]
```diff
@@ -776,7 +782,6 @@ router.postAsync("/streamer", ensure.isStreamer, async (req, res) => {
  * Removes the current streamer
  * Params:
  *  - match: the _id of the match
- *  - tourney: identifier for the tournament
```
Comment (cory2067) [inline server/api.ts:779]: oops

**Review by cory2067 — APPROVED** [review]
nice

File: server/api.ts [hunk 3]
```diff
@@ -552,29 +552,35 @@ router.postAsync("/match", ensure.isAdmin, async (req, res) => {
...
+    // Make sure mod is valid, if it's not default to nm
+    if (req.body.mod != "DT") { req.body.mod = "NM" }
+    match[`warmup${req.body.playerNo}`] = await parseWarmup(req.body.warmup, req.body.mod || "NM", match.tourney);
+    match[`warmup${req.body.playerNo}Mod`] = req.body.mod;
   } catch (e) {
     res.status(400).send(e.message);
     return;
   }
-  logger.info(`${req.user.username} submitted warmup ${req.body.warmup}`);
+  logger.info(`${req.user?.username} submitted warmup ${req.body.warmup}`);
```
Comment (cory2067) [inline server/api.ts:583]: btw there is also this helper from util.ts you can use if you don't want to keep checking user for undefined
```
const user = assertUser(req);
```

File: server/api.ts (hunk 2 above)
Comment (vunyunt) [reply server/api.ts]: oops fixed in bc1e56a1a715dbb05a0259c6d8b85e0afb8d2346

File: server/api.ts (hunk 3 above)
Comment (vunyunt) [reply server/api.ts]: Ahh didn't know, applied in cb935bf6da1ca23fe77b22eff90b5ea1215b2199

Though I kept the permission check functions to allow undefined user being passed in

**Review by cory2067 — APPROVED** [review]

**Pull request merged**